bfa-automation-java/src/main/java/com/bfa/core/page/Page.java

```diff
@@ -1,17 +1,29 @@
 package com.bfa.core.page;
 
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import static com.bfa.core.util.LocatorUtils.getLocatorByKey;
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated;
 
 /**
  * Created by ninhdoan on 12/24/16.
  */
 public class Page {
 
+   public static final String DEFAULT_WAIT_TIME_IN_SECONDS_KEY = "webdriver.defaultWaitTimeInSeconds";
+
    protected final WebDriver webDriver;
 
    public Page(WebDriver webDriver) {
       this.webDriver = webDriver;
+   }
 
+   public WebElement waitForWebElementToBeVisible(String webElementLocatorKey, long waitTimeInSeconds) {
+      WebDriverWait webDriverWait = new WebDriverWait(this.webDriver, waitTimeInSeconds);
+      WebElement visibleWebElement = webDriverWait.until(visibilityOfElementLocated(getLocatorByKey(webElementLocatorKey)));
+      return visibleWebElement;
```

Comment:
This is in library layer, then the method would be called many times. I wonder why we need to ask caller providing timeout so many times with this design. In my opinion, I prefer the optional parameter for timeout, and if user does not provide timeout, we use the default timeout (for example, 30 seconds) which is configured in the config file.